### PR TITLE
Replace default-xalan xslt processor with Saxon-He

### PR DESCRIPTION
**Pull request by sahanHe**
## Examples
Fixes:[ballerina-standard-library/issues/3568](https://github.com/ballerina-platform/ballerina-standard-library/issues/3568) and Related to issue : [3462](https://github.com/ballerina-platform/ballerina-standard-library/issues/3462)
## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [ ] Added tests
- [ ] Updated the spec
- [ ] Checked native-image compatibility

### CODE REVIEW

File: native/src/main/java/io/ballerina/stdlib/xslt/XsltTransformer.java [hunk 1]
```diff
@@ -79,11 +82,13 @@ public static Object transform(BXml xmlInput, BXml xslInput) {
             StringWriter stringWriter = new StringWriter();
             StreamResult streamResult = new StreamResult(stringWriter);
 
-            Transformer transformer = TransformerFactory.newInstance().newTransformer(xslSource);
+            Transformer transformer = new BasicTransformerFactory().newInstance().newTransformer(xslSource);
+            errStream.println(transformer);
```
Comment (kalaiyarasiganeshalingam) [inline native/src/main/java/io/ballerina/stdlib/xslt/XsltTransformer.java:86]: Is it necessary?

File: ballerina/Ballerina.toml [hunk 2]
```diff
@@ -18,3 +18,11 @@ path = "../native/build/libs/xslt-native-2.3.0-SNAPSHOT.jar"
 [[platform.java11.dependency]]
 path = "./lib/jakarta.activation-1.2.2.jar"
 version = "1.2.2"
```
Comment (daneshk) [inline ballerina/Ballerina.toml:20]: Do we need jakarta dependency?

File: native/src/main/java/io/ballerina/stdlib/xslt/XsltTransformer.java [hunk 3]
```diff
@@ -52,11 +53,13 @@
 public class XsltTransformer {
 
     private static final Logger log = LoggerFactory.getLogger(XsltTransformer.class);
+    private static final PrintStream errStream = System.err;
     private static final String OPERATION = "Failed to perform XSL transformation: ";
 
     public static Object transform(BXml xmlInput, BXml xslInput) {
 
         try {
+            //System.setProperty("javax.xml.transform.TransformerFactory", "net.sf.saxon.BasicTransformerFactory");
```
Comment (daneshk) [inline native/src/main/java/io/ballerina/stdlib/xslt/XsltTransformer.java:62]: Let's remove commented lines

Comment (sahanHe) [reply native/src/main/java/io/ballerina/stdlib/xslt/XsltTransformer.java]: Updated in [cfe19f6](https://github.com/ballerina-platform/module-ballerina-xslt/pull/427/commits/cfe19f6c1557bd6f2848b10770c41649fc72e896)

File: native/src/main/java/io/ballerina/stdlib/xslt/XsltTransformer.java (hunk 1 above)
Comment (sahanHe) [reply native/src/main/java/io/ballerina/stdlib/xslt/XsltTransformer.java]: Updated in [cfe19f6](https://github.com/ballerina-platform/module-ballerina-xslt/pull/427/commits/cfe19f6c1557bd6f2848b10770c41649fc72e896)

File: ballerina/Ballerina.toml (hunk 2 above)
Comment (sahanHe) [reply ballerina/Ballerina.toml:20]: Yes, it is required dependancy for the code. Otherwise we get `ClassNotFoundException` in runtime

**Review by daneshk — APPROVED** [review]

**Pull request merged**